.github/workflows/release.yml

```diff
@@ -141,3 +154,7 @@ jobs:
           if [ -n "${GPG_KEY_FILE}" ]; then
             rm -rf "${GPG_KEY_FILE}"
           fi
+
+      - name: "Check free space on runner"
```

Comment:
This will let us know if we're running out of space on the Linux runner in the future. We could split the matrix by architecture as well; that'd create even more parallel jobs; IMO there's a bit of value in avoiding the shared overhead of getting to a usable state, but there's also not much compilation cache that can be shared otherwise since the architecture is different.

My 2c.

After this stage, we could also have a validation stage that automatically tests the built artifacts for us (signature validation, container fetches, &c), but that's a problem for later.